src/librustc_target/spec/riscv64imc_unknown_none_elf.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use spec::{LinkerFlavor, LldFlavor, PanicStrategy,
+           Target, TargetOptions, TargetResult};
+
+pub fn target() -> TargetResult {
+    Ok(Target {
+        data_layout: "e-m:e-i64:64-n64-S128".to_string(),
```

Comment:
Is this the correct data layout to pass to LLVM? I read https://llvm.org/docs/LangRef.html#data-layout but I'm not 100% sure I got everything correct

Reply:
https://github.com/lowRISC/riscv-llvm/blob/3a81c43b0294d6677f589a8a26137279943c1577/clang/0002-RISCV-Add-the-RISCV-target-and-compiler-driver.patch#L264

Reply:
Ok. That version seems to be equivalent, since `p:64:64:64` and `f128:128:128` are default. Though I don't quite understand why the defaults have an extra section to them... Should I copy over the configuration from lowRISC?

Reply:
Currently rustc verifies that the data layout passed to LLVM matches the default data-layout for the target. IIRC it is an ICE or an error otherwise, don’t remember which exactly.

So you have to have exactly the same string as the backend you’re hoping to use.

Reply:
Maybe it doesn't matter, but I think being consistent with the clang driver isn't a bad policy